File: backend/internal/service/system/admin/v1/base_job.go
package admin

import (
	"context"
	"fmt"

	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	commonv1 "github.com/liujitcn/kratos-core/api/gen/go/common/v1"
	"github.com/liujitcn/kratos-core/errorsx"

	"github.com/go-kratos/kratos/v3/log"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseJobService Admin定时任务服务
type BaseJobService struct {
	adminv1.UnimplementedBaseJobServiceServer
	baseJobCase *biz.BaseJobCase
}

// NewBaseJobService 创建Admin定时任务服务
func NewBaseJobService(baseJobCase *biz.BaseJobCase) *BaseJobService {
	return &BaseJobService{baseJobCase: baseJobCase}
}

// OptionBaseJob 查询定时任务下拉选择
func (s *BaseJobService) OptionBaseJob(ctx context.Context, req *adminv1.OptionBaseJobRequest) (*commonv1.SelectOptionResponse, error) {
	list, err := s.baseJobCase.OptionBaseJob(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("OptionBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "查询定时任务下拉列表失败")
	}
	return list, nil
}

// PageBaseJob 查询定时任务分页列表
func (s *BaseJobService) PageBaseJob(ctx context.Context, req *adminv1.PageBaseJobRequest) (*adminv1.PageBaseJobResponse, error) {
	page, err := s.baseJobCase.PageBaseJob(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "查询定时任务分页列表失败")
	}
	return page, nil
}

// GetBaseJob 查询定时任务
func (s *BaseJobService) GetBaseJob(ctx context.Context, req *adminv1.GetBaseJobRequest) (*adminv1.BaseJobForm, error) {
	baseJob, err := s.baseJobCase.GetBaseJob(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "查询定时任务失败")
	}
	return baseJob, nil
}

// CreateBaseJob 创建定时任务
func (s *BaseJobService) CreateBaseJob(ctx context.Context, req *adminv1.CreateBaseJobRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.CreateBaseJob(ctx, req.GetBaseJob())
	if err != nil {
		log.Error(fmt.Sprintf("CreateBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "创建定时任务失败")
	}
	return new(emptypb.Empty), nil
}

// UpdateBaseJob 更新定时任务
func (s *BaseJobService) UpdateBaseJob(ctx context.Context, req *adminv1.UpdateBaseJobRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.UpdateBaseJob(ctx, req.GetBaseJob())
	if err != nil {
		log.Error(fmt.Sprintf("UpdateBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "更新定时任务失败")
	}
	return new(emptypb.Empty), nil
}

// DeleteBaseJob 删除定时任务
func (s *BaseJobService) DeleteBaseJob(ctx context.Context, req *adminv1.DeleteBaseJobRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.DeleteBaseJob(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("DeleteBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "删除定时任务失败")
	}
	return new(emptypb.Empty), nil
}

// SetBaseJobStatus 设置状态
func (s *BaseJobService) SetBaseJobStatus(ctx context.Context, req *adminv1.SetBaseJobStatusRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.SetBaseJobStatus(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("SetBaseJobStatus %v", err))
		return nil, errorsx.WrapInternal(err, "设置状态失败")
	}
	return new(emptypb.Empty), nil
}

// StartBaseJob 启动任务
func (s *BaseJobService) StartBaseJob(ctx context.Context, req *adminv1.StartBaseJobRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.StartBaseJob(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("StartBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "启动任务失败")
	}
	return new(emptypb.Empty), nil
}

// StopBaseJob 停止任务
func (s *BaseJobService) StopBaseJob(ctx context.Context, req *adminv1.StopBaseJobRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.StopBaseJob(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("StopBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "停止任务失败")
	}
	return new(emptypb.Empty), nil
}

// ExecuteBaseJob 执行任务
func (s *BaseJobService) ExecuteBaseJob(ctx context.Context, req *adminv1.ExecuteBaseJobRequest) (*emptypb.Empty, error) {
	err := s.baseJobCase.ExecuteBaseJob(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("ExecuteBaseJob %v", err))
		return nil, errorsx.WrapInternal(err, "执行任务失败")
	}
	return new(emptypb.Empty), nil
}

File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_job.ts
import service from "@liujitcn/kratos-admin-core/request";
import {
  type BaseJobForm,
  type BaseJobService,
  type CreateBaseJobRequest,
  type DeleteBaseJobRequest,
  type ExecuteBaseJobRequest,
  type GetBaseJobRequest,
  type OptionBaseJobRequest,
  type PageBaseJobRequest,
  type PageBaseJobResponse,
  type SetBaseJobStatusRequest,
  type StartBaseJobRequest,
  type StopBaseJobRequest,
  type UpdateBaseJobRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_job";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";
import type { SelectOptionResponse } from "@liujitcn/kratos-admin-system/rpc/common/v1/common";

const BASE_JOB_URL = "/v1/admin/base/job";

/** Admin定时任务服务 */
export class BaseJobServiceImpl implements BaseJobService {
  /** 查询定时任务下拉选择 */
  OptionBaseJob(request: OptionBaseJobRequest): Promise<SelectOptionResponse> {
    return service<OptionBaseJobRequest, SelectOptionResponse>({
      url: `${BASE_JOB_URL}/option`,
      method: "get",
      params: request
    });
  }

  /** 查询定时任务分页列表 */
  PageBaseJob(request: PageBaseJobRequest): Promise<PageBaseJobResponse> {
    return service<PageBaseJobRequest, PageBaseJobResponse>({
      url: `${BASE_JOB_URL}`,
      method: "get",
      params: request
    });
  }

  /** 查询定时任务 */
  GetBaseJob(request: GetBaseJobRequest): Promise<BaseJobForm> {
    return service<GetBaseJobRequest, BaseJobForm>({
      url: `${BASE_JOB_URL}/${request.id}`,
      method: "get"
    });
  }

  /** 创建定时任务 */
  CreateBaseJob(request: CreateBaseJobRequest): Promise<Empty> {
    return service<BaseJobForm | undefined, Empty>({
      url: `${BASE_JOB_URL}`,
      method: "post",
      data: request.base_job
    });
  }

  /** 更新定时任务 */
  UpdateBaseJob(request: UpdateBaseJobRequest): Promise<Empty> {
    return service<BaseJobForm | undefined, Empty>({
      url: `${BASE_JOB_URL}/${request.base_job?.id ?? ""}`,
      method: "put",
      data: request.base_job
    });
  }

  /** 删除定时任务 */
  DeleteBaseJob(request: DeleteBaseJobRequest): Promise<Empty> {
    return service<DeleteBaseJobRequest, Empty>({
      url: `${BASE_JOB_URL}/${request.id}`,
      method: "delete"
    });
  }

  /** 设置状态 */
  SetBaseJobStatus(request: SetBaseJobStatusRequest): Promise<Empty> {
    return service<SetBaseJobStatusRequest, Empty>({
      url: `${BASE_JOB_URL}/${request.id}/status`,
      method: "put",
      data: request
    });
  }

  /** 启动任务 */
  StartBaseJob(request: StartBaseJobRequest): Promise<Empty> {
    return service<StartBaseJobRequest, Empty>({
      url: `${BASE_JOB_URL}/${request.id}/running`,
      method: "put",
      data: request
    });
  }

  /** 停止任务 */
  StopBaseJob(request: StopBaseJobRequest): Promise<Empty> {
    return service<StopBaseJobRequest, Empty>({
      url: `${BASE_JOB_URL}/${request.id}/running`,
      method: "delete",
      data: request
    });
  }

  /** 执行任务 */
  ExecuteBaseJob(request: ExecuteBaseJobRequest): Promise<Empty> {
    return service<ExecuteBaseJobRequest, Empty>({
      url: `${BASE_JOB_URL}/${request.id}/execution`,
      method: "post",
      data: request
    });
  }

}

export const defBaseJobService = new BaseJobServiceImpl();
